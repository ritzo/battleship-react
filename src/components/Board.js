import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import Square from './Square';
import { BOARD_TYPES } from '../constants/Constants';

import './Board.css';

function renderSquare(squares, handleClick, i) {
  return (
    <Square
      key={i}
      value={squares[i]}
      onClick={() => handleClick(i)}
    />
  );
}

function createTable(squares, handleClick) {
  const table = [];

  for (let i = 0; i < 10; i += 1) {
    const children = [];

    for (let j = 0; j < 10; j += 1) {
      const index = (i * 10) + j;
      children.push(renderSquare(squares, handleClick, index));
    }

    table.push(<div key={i} className="board-row">{children}</div>);
  }
  return table;
}

class Board extends React.Component {
  render() {
    const {
      title, type, playerBoard, oponentBoard,
    } = this.props;
    let boardClass = '';
    let squares = [];

    switch (type) {
      case BOARD_TYPES.EDITION:
        boardClass = 'edition';
        squares = playerBoard;
        break;

      case BOARD_TYPES.PLAYER:
        boardClass = 'player';
        squares = playerBoard;
        break;

      case BOARD_TYPES.OPONENT:
        boardClass = 'oponent';
        squares = oponentBoard;
        break;

      default:
        // Invalid board type
        break;
    }

    this.fun = (index) => {
      // eslint-disable-next-line no-alert
      alert(`asdsad${index}`);
    };

    return (
      <div className={boardClass}>
        { type }
        <div className="status">{ title }</div>
        { createTable(squares, this.fun) }
      </div>
    );
  }
}

Board.propTypes = {
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  playerBoard: PropTypes.arrayOf(PropTypes.string).isRequired,
  oponentBoard: PropTypes.arrayOf(PropTypes.string).isRequired,
};

function mapStateToProps(state) {
  return {
    playerBoard: state.playerBoard,
    oponentBoard: state.oponentBoard,
    playerBoardLast: state.playerBoardLast,
    oponentBoardLast: state.oponentBoardLast,
  };
}

export default connect(mapStateToProps)(Board);
