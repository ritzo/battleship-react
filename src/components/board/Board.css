ol, ul {
  padding-left: 30px;
}

.board-row:after {
  clear: both;
  content: "";
  display: table;
}

.is-next .table-container {
  border: 3px solid rgb(114, 66, 66);
  width: fit-content;
}

.table-container {
  border: 3px solid transparent;
  width: fit-content;
  margin: auto;
}

.status {
  margin-bottom: 10px;
  text-align: center;
  font-size: 20px;
  font-weight: bold;
}
