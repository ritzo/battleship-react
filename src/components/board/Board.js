import React from 'react';
import PropTypes from 'prop-types';

import SquareEditorContainer from '../square/SquareEditorContainer';
import SquarePlayerContainer from '../square/SquarePlayerContainer';
import SquareOponentContainer from '../square/SquareOponentContainer';
import NextShipContainer from '../nextShip/NextShipContainer';
import { BOARD_TYPES } from '../../constants/Constants';

import './Board.css';

function renderSquare(x, y, boardType) {
  switch (boardType) {
    case BOARD_TYPES.EDITION:
      return (
        <SquareEditorContainer
          key={x}
          x={x}
          y={y}
        />
      );
    case BOARD_TYPES.PLAYER:
      return (
        <SquarePlayerContainer
          key={x}
          x={x}
          y={y}
        />
      );
    case BOARD_TYPES.OPONENT:
      return (
        <SquareOponentContainer
          key={x}
          x={x}
          y={y}
        />
      );
    default:
      return null;
  }
}

function createTable(boardType) {
  const table = [];

  for (let y = 0; y < 10; y += 1) {
    const children = [];
    for (let x = 0; x < 10; x += 1) {
      children.push(renderSquare(x, y, boardType));
    }
    // The elements in the matrix won't change positions nor new items will be added
    // eslint-disable-next-line react/no-array-index-key
    table.push(<div key={y} className="board-row">{children}</div>);
  }

  return table;
}

class Board extends React.Component {
  render() {
    const {
      title, type, isNext,
    } = this.props;

    const boardClass = `${type} ${isNext ? 'is-next' : ''}`;

    this.renderEdition = () => {
      if (BOARD_TYPES.EDITION !== type) {
        return null;
      }
      return (
        <div>
          <p className="edition-hint">Left click to add a ship horizontally, right click to add it vertically.</p>
          <NextShipContainer />
        </div>
      );
    };

    return (
      <div className={boardClass}>
        <div className="status">{ title }</div>
        <div className="table-container">
          { createTable(type) }
        </div>
        { this.renderEdition() }
      </div>
    );
  }
}

Board.propTypes = {
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  isNext: PropTypes.bool.isRequired,
};

export default Board;
