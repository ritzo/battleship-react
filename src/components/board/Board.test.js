import React from 'react';
import { shallow } from 'enzyme';

import Board from './Board';

describe('Board', () => {
  it('Snapshot', () => {
    const enzymeWrapper = shallow(<Board title="test board" type="someType" isNext={false} />);

    expect(enzymeWrapper).toMatchSnapshot();
  });
});
