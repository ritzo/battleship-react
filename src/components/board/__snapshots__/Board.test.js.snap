// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board Snapshot 1`] = `
<div
  className="someType "
>
  <div
    className="status"
  >
    test board
  </div>
  <div
    className="table-container"
  >
    <div
      className="board-row"
      key="0"
    />
    <div
      className="board-row"
      key="1"
    />
    <div
      className="board-row"
      key="2"
    />
    <div
      className="board-row"
      key="3"
    />
    <div
      className="board-row"
      key="4"
    />
    <div
      className="board-row"
      key="5"
    />
    <div
      className="board-row"
      key="6"
    />
    <div
      className="board-row"
      key="7"
    />
    <div
      className="board-row"
      key="8"
    />
    <div
      className="board-row"
      key="9"
    />
  </div>
</div>
`;
